muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/ForumRESTService.java

```diff
@@ -749,8 +760,10 @@ public Response createReply(@PathParam ("AREAID") Long areaId, @PathParam ("THRE
         return Response.status(Status.NOT_FOUND).entity("Forum thread not found from the specified area").build();
       }
       
-      if (forumThread.getLocked()) {
-        return Response.status(Status.BAD_REQUEST).entity("Forum thread is locked").build();
+      if (forumThread.getLock() != null) {
```

Comment:
Not needed; `forumThread.getLock() == LockForumThread.ALL` is inherently null-safe.